cna-memory-considerate-animation/pages/_app.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { AnimatePresence } from 'framer-motion';
+
+import Layout from '../components/Layout';
+import AnimationEmulationContext from '../components/AnimationEmulationContext';
+import { useMemoryStatus } from '../utils/hooks';
+import { CLIENT_HINT_MEMORY_LIMIT } from '../config';
+
+const MyApp = ({ Component, pageProps, router }) => {
+  const [manualEnabled, setManualEnabled] = useState(false);
+  const [isAnimationOn, setIsAnimationOn] = useState(true);
+
+  const { clientHintDeviceMemory, memoryStatus } = pageProps;
+  let overLoaded;
+  if (clientHintDeviceMemory) {
+    overLoaded = clientHintDeviceMemory < CLIENT_HINT_MEMORY_LIMIT;
+    console.log('[_app MyApp] Client Hint Device Memory based Memory Overloaded => ', overLoaded);
+  } else {
+    overLoaded = memoryStatus.overLoaded;
+    console.log('[_app MyApp] Memory React Hook based Memory Overloaded => ', overLoaded);
+  }
```

Comment:
Memory status evaluation conditional for server side rendering and client side rendering.